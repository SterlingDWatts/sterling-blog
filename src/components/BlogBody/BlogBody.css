.BlogBody {
  line-height: 1.5em;
  width: 100%;
}

.BlogBody__title {
  font-weight: 600;
  font-size: 24px;
  margin: 32px 0px;
}

.BlogBody__pic {
  width: 100%;
  height: calc((100vw - 32px) / 900 * 510);
  background-position: center;
  background-size: cover;
}

.BlogBody__date_and_views {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  margin-bottom: 32px;
  font-size: 13px;
  color: grey;
}

.BlogBody__content strong {
  font-weight: 600;
}

.BlogBody__content em {
  font-style: italic;
}

.BlogBody__content sub {
  vertical-align: sub;
  font-size: smaller;
}

.BlogBody__content sup {
  vertical-align: super;
  font-size: smaller;
}

.BlogBody__content h3 {
  font-size: 24px;
  font-weight: 600;
}

.BlogBody--color-splash {
  color: #fa813d;
}

.BlogBody__buttons {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;
  height: 157.5px;
}

@media screen and (min-width: 650px) {
  .BlogBody__buttons {
    flex-direction: row;
    max-width: 700px;
  }
}
