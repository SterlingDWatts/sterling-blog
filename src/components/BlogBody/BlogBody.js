import React, { Component } from "react";
import { NiceDate } from "../Utils/Utils";
import BlogBodyContext from "../../contexts/BlogPageContext";
import TokenService from "../../services/token-service";
import "./BlogBody.css";

class BlogBody extends Component {
  static contextType = BlogBodyContext;

  static defaultProps = {
    onDelete: () => {},
    onEdit: () => {}
  };

  renderButtons = () => {
    return (
      <>
        <button className="BlogBody__delete_button" onClick={this.props.onEdit}>
          Edit Blog
        </button>
        <button
          className="BlogBody__delete_button"
          onClick={this.props.onDelete}
        >
          Delete Blog
        </button>
      </>
    );
  };

  render() {
    const {
      title,
      picture,
      content,
      author,
      date_created,
      number_of_views
    } = this.context.blog;
    return (
      <main className="BlogBody">
        <header>
          <h2 className="BlogBody__title">{title}</h2>
          <h3 className="BlogBody__author">
            {author.first_name + " " + author.last_name}
          </h3>
        </header>
        <div
          className="BlogBody__pic"
          style={{ backgroundImage: "url('" + picture + "')" }}
        ></div>
        <div className="BlogBody__date_and_views">
          <div className="BlogBody__date">
            {date_created && <NiceDate date={date_created} />}
          </div>
          <div className="BlogBody__views">{number_of_views} views</div>
        </div>
        <div
          dangerouslySetInnerHTML={{ __html: content }}
          className="BlogBody__content"
        />
        <div className="BlogBody__buttons">
          {TokenService.hasAuthToken() &&
            TokenService.hasPrivileges(author.id) &&
            this.renderButtons()}
        </div>
      </main>
    );
  }
}

export default BlogBody;
