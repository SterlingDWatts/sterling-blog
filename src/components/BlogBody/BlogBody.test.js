import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import BlogBody from "./BlogBody";

describe("BlogBody Component", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<BlogBody />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders the UI as expected", () => {
    const tree = renderer.create(<BlogBody />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
