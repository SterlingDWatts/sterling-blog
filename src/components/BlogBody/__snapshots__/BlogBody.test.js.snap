// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogBody Component renders the UI as expected 1`] = `
<main
  className="BlogBody"
>
  <header>
    <h2
      className="BlogBody__title"
    />
    <h3
      className="BlogBody__author"
    >
      undefined undefined
    </h3>
  </header>
  <div
    className="BlogBody__pic"
    style={
      Object {
        "backgroundImage": "url('undefined')",
      }
    }
  />
  <div
    className="BlogBody__date_and_views"
  >
    <div
      className="BlogBody__date"
    />
    <div
      className="BlogBody__views"
    >
       views
    </div>
  </div>
  <div
    className="BlogBody__content"
    dangerouslySetInnerHTML={
      Object {
        "__html": undefined,
      }
    }
  />
  <div
    className="BlogBody__buttons"
  />
</main>
`;
